Modules

cxvint opened the pull request.

cxvint committed:
Lab_Weather

cxvint committed:
Lab_Weather

	new file:   AJAX/LAB/app.js
	new file:   AJAX/LAB/by-cities.json
	new file:   AJAX/LAB/images/clear.png
	new file:   AJAX/LAB/images/clouds.png
	new file:   AJAX/LAB/images/rain.png
	new file:   AJAX/LAB/images/sun.png
	new file:   AJAX/LAB/index.html
	new file:   AJAX/LAB/style.css

cxvint committed:
Merge branch 'ajaxLab' of https://github.com/cxvint/HW into ajaxLab

cxvint committed:
modified:   AJAX/LAB/app.js

cxvint committed:
modified:   AJAX/LAB/app.js

cxvint committed:
hw_modules

NooNoo1337 commented on AJAX/LAB/app.js:
```diff
@@ -0,0 +1,123 @@
+    if (e.target.readyState !== 4) {
+      return;
+    }
+    if (xhr.status != 200) {
```
**suggestion:** !==

Лучше делать всегда безопасно и избегать приведение типов

cxvint replied:
Такой пример Саша на лекции показал. Так, а что сюда лучше вставить, вместо !==?

NooNoo1337 replied:
У тебя !=, а надо как раз  !==

NooNoo1337 commented on Modules/hw/app.js:
```diff
@@ -0,0 +1,33 @@
+}
+
+function showPasswordStrengthInColor(e) {
+  const target = e.target;
```
**nitpick:** const { target } = e;

NooNoo1337 commented on Modules/hw/app.js:
```diff
@@ -0,0 +1,33 @@
+      return 'rgba(255, 255, 0, 0.7)';
+    case 3:
+      return 'rgba(150, 255, 0, 0.7)';
+    case 4:
```
**issue:** Не обработан default case

cxvint replied:
+ ...на developer.mozilla.org, почему-то пишут, что случай default  - не является строгим правилом.

NooNoo1337 commented on AJAX/LAB/style.css:
```diff
@@ -0,0 +1,37 @@
+}
+.new-cities {
+  background-color: rgb(160, 0, 160);
+}
```
**nitpiсk:** Почему нет пробелов между классами?

NooNoo1337 commented on AJAX/LAB/app.js:
```diff
@@ -0,0 +1,123 @@
+
+      weatherImg.src = 'images/' + data.weather[0].main.toLowerCase() + '.png';
+
+      let temp = Math.round(data.main.temp - 273);
```
**suggestion:** Получение температуры пол логике - отдельный функционал от погоды. Можно вынести в отдельную функцию

NooNoo1337 commented on AJAX/LAB/app.js:
```diff
@@ -0,0 +1,123 @@
+      return;
+    }
+    if (xhr.status != 200) {
+      console.log(`${xhr.status} + ': ' + ${xhr.statusText}`);
```
**nitpick:** Клей строки через бектики ``

NooNoo1337 commented on AJAX/LAB/app.js:
```diff
@@ -0,0 +1,123 @@
+
+    let newArrayCities;
+
+    function getArrCities(arrCities) {
```
**suggestion:** Это можно вынести из request

cxvint committed:
Modules

NooNoo1337 commented on Modules/hw/app.js:
```diff
@@ -0,0 +1,35 @@
+const catAge = document.querySelector('.cat-age');
+
+humanAge.addEventListener('password', () => {
+  let result = Number(humanAge.value);
```
**question:** Почему let?

NooNoo1337 commented on AJAX/LAB/style.css:
```diff
@@ -0,0 +1,37 @@
+  width: 190px;
+
+  box-shadow: 0.4em 0.4em 5px rgba(116, 211, 255, 0.5);
+}
```
**nitpick:** Почему классы слиплись? Нужны отступы между классами

NooNoo1337 commented:
Так, а что в данном мр делают изменения, которые относятся к домашке с погодой?

cxvint committed:
new file:   .gitignore
	deleted:    AJAX/LAB/app.js
	deleted:    AJAX/LAB/by-cities.json
	deleted:    AJAX/LAB/images/clear.png
	deleted:    AJAX/LAB/images/clouds.png
	deleted:    AJAX/LAB/images/rain.png
	deleted:    AJAX/LAB/images/sun.png
	deleted:    AJAX/LAB/index.html
	deleted:    AJAX/LAB/style.css
	deleted:    AJAX/SPA/about.html
	deleted:    AJAX/SPA/about.json
	deleted:    AJAX/SPA/app.js
	deleted:    AJAX/SPA/contacts.html
	deleted:    AJAX/SPA/contacts.json
	deleted:    AJAX/SPA/index.html
	deleted:    AJAX/SPA/index.json
	deleted:    AJAX/dynamicPageData/index.html
	deleted:    AJAX/dynamicPageData/index.json
	modified:   Modules/hw/app.js

cxvint committed:
fix modules2

cxvint committed:
new file:   .gitignore

NooNoo1337 commented on Modules/hw/app.js:
```diff
@@ -0,0 +1,35 @@
+//----------------------------password------------------------------
+const passwordInput = document.querySelector('.password');
+
+function transformScoreToColor(number) {
```
**suggestion:** Гораздо более быстрее будет обрабатываться код, если ты сделаешь объект, ключами которого будет число (0-4), а значениями цвет.

NooNoo1337 reviewed (approved):
Approve

NooNoo1337 commented:
Approve
